Sources/LiveViewNative/Modifiers/Text Input and Output Modifiers/FontDesignModifier.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  FontDesignModifier.swift
+// LiveViewNative
+//
+//  Created by May Matyi on 4/24/23.
+//
+
+import SwiftUI
+
+/// Applies a font design to any element.
+///
+/// <Text modifiers={font_design(@native, design: :default)}>This text is default</Text>
+/// <Text modifiers={font_design(@native, design: :monospaced)}>This text is monospaced</Text>
+/// <Text modifiers={font_design(@native, design: :rounded)}>This text is rounded</Text>
+/// <Text modifiers={font_design(@native, design: :serif)}>This text is serif</Text>
+///
+#if swift(>=5.8)
+@_documentation(visibility: public)
+#endif
+struct FontDesignModifier: ViewModifier, Decodable {
+    /// The font design to apply to the view.
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    private var design: Font.Design
+    
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let designString = try container.decode(String.self, forKey: .design)
+
+        switch designString {
+        case "default":
+            design = .default
+        case "monospaced":
+            design = .monospaced
+        case "rounded":
+            design = .rounded
+        case "serif":
+            design = .serif
+        default:
+            throw DecodingError.dataCorruptedError(forKey: .design, in: container, debugDescription: "expected valid value for Font.Design");
+        }
+    }
+
+    func body(content: Content) -> some View {
+        content.font(.system(size: UIFont.labelFontSize, design: design))
```

Comment:
This should directly use the `.fontDesign` modifier, since this will override the size if it's set elsewhere and will break dynamic type

Reply:
@shadowfacts Fixed. It seems that `.fontDesign` is only available on iOS 16.1 or later so I had to use an `#available` version check. I'm not sure how to handle the fallback in a way that doesn't override the font size if its already set, so I ended up just ignoring it on older versions (feel free to make a suggestion for how to handle that if its something we want to do)

Reply:
I agree. It'd be nice if we could mark the the modifier type with `@available` so the docs reflect the actual availability, but the builtin registry generator doesn't support it at the moment (because the availability check would move into the generated code).

Reply:
@shadowfacts I think this is addressed by https://github.com/liveview-native/liveview-client-swiftui/pull/868 and out of scope for this PR anyway so marking this as resolved 🙂